pskb_website/views.py

```diff
@@ -546,20 +546,22 @@ def partner(article_path):
                            disclaimer=True)
 
 
-@app.route('/save/', methods=['POST'])
+@app.route('/api/save/', methods=['POST'])
 @login_required
-def save():
-    """Form POST save"""
+def api_save():
+    """Api: POST /api/save {path:'', title: '', sha:'', original_stack: '', content: '', stacks: []}"""
 
     g.slack_url = SLACK_URL
 
     user = models.find_user()
     if user is None:
-        flash('Cannot save unless logged in', category='error')
-        return render_template('index.html'), 404
+        redirect_to = url_for('index')
+        data = {'error': 'Cannot save unless logged in', 'redirect': redirect_to}
+        return Response(response=json.dumps(data), status=401, mimetype='application/json')
 
     if user.email is None:
         flash('Unable to read email address from Github API to properly attribute your commit to your account. Please make sure you have authorized the application to access your email.', category='warning')
+        # FIXME: stop using flash
```

Comment:
What would you suggest we do here? Maybe save off this message and return it with response?